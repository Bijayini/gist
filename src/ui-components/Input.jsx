import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import theme from '../lib/theme';

const Input = ({ placeholder, onInputChange, value, id, hasError }) => (
  <Fragment>
    <input
      type="text"
      id={id}
      value={value}
      onChange={event => onInputChange(event)}
      placeholder={placeholder}
      className={hasError && 'hasError'}
      autoComplete="off"
    />
    <style jsx="true">
      {`
        input {
          width: 100%;
          border: 1px solid ${theme.border};
          border-radius: 3px;
          padding: 5px 10px;
          font-size: 22px;
          outline: none;
          font-family: Lato;
          font-weight: 900;
        }
        input.hasError {
          border-color: ${theme.warningMessageColor};
        }
        input::placeholder {
          font-family: Lato;
          font-size: 16px;
          color: ${theme.grey4};
        }
      `}
    </style>
  </Fragment>
);

Input.defaultProps = {
  id: '',
  hasError: false,
};

Input.propTypes = {
  placeholder: PropTypes.string.isRequired,
  onInputChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
  id: PropTypes.string,
  hasError: PropTypes.bool,
};

export default Input;
