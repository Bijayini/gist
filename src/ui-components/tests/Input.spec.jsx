import React from 'react';
import { shallow } from 'enzyme';

import Input from '../Input';

describe('<Input />', () => {
  const props = {
    placeholder: 'some placeholder',
    onInputChange: jest.fn(),
    value: 'some value',
    border: '#f0f0f0f',
  };
  it('should match with snapshot', () => {
    const component = shallow(<Input {...props} />);

    expect(component.debug()).toMatchSnapshot();
  });

  it('should call handleChange on change in input', () => {
    const component = shallow(<Input {...props} />);
    const input = component.find('input');

    input.simulate('change', { target: { value: 5 } });

    expect(props.onInputChange).toHaveBeenCalledWith({ target: { value: 5 } });
  });

  it('should render has error when there is an error', () => {
    props.hasError = true;
    const component = shallow(<Input {...props} />);
    const inputWithError = component.find('.hasError');

    expect(inputWithError).toHaveLength(1);
  });
});
