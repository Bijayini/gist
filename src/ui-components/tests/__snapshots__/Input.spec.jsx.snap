// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should match with snapshot 1`] = `
"<Fragment>
  <input type=\\"text\\" id=\\"\\" value=\\"some value\\" onChange={[Function: onChange]} placeholder=\\"some placeholder\\" autoComplete=\\"off\\" className=\\"jsx-1118511121 \\" />
  <JSXStyle id=\\"365830188\\" dynamic={{...}}>
    input.__jsx-style-dynamic-selector{width:100%;border:1px solid #c8c8c8;border-radius:3px;padding:5px 10px;font-size:22px;outline:none;font-family:Lato;font-weight:900;}input.hasError.__jsx-style-dynamic-selector{border-color:#be0000;}input.__jsx-style-dynamic-selector::-webkit-input-placeholder{font-family:Lato;font-size:16px;color:#d8d8d8;}input.__jsx-style-dynamic-selector::-moz-placeholder{font-family:Lato;font-size:16px;color:#d8d8d8;}input.__jsx-style-dynamic-selector:-ms-input-placeholder{font-family:Lato;font-size:16px;color:#d8d8d8;}input.__jsx-style-dynamic-selector::placeholder{font-family:Lato;font-size:16px;color:#d8d8d8;}
  </JSXStyle>
</Fragment>"
`;
